include/xplot/xaxes.hpp

```diff
@@ -67,6 +68,39 @@ namespace xpl
 
     using axis = xw::xmaterialize<xaxis>;
 
+    /***************************
+    * xcolor_axis declaration *
+    ***************************/
+
+    template<class D>
+    class xcolor_axis: public xaxis<D>
+    {
+    public:
+
+        using base_type = xaxis<D>;
+        using derived_type = D;
+        using scale_type = xw::xholder<xscale>;
+
+        template <class S>
+        xcolor_axis(const xscale<S>&);
```

Comment:
This is ok, although we can probably constrain it more by only taking `xcolor_scale` instead of `xscale`.

Reply:
You're right. I will do that and if you are ok with the rest of this PR, you can merge it.